src/test/java/org/apache/commons/text/CharacterPredicatesTest.java

```diff
@@ -44,4 +44,77 @@ public void testDigits() {
         assertFalse(CharacterPredicates.DIGITS.test('.'));
         assertFalse(CharacterPredicates.DIGITS.test('L'));
     }
+
+    @Test
+    public void testArabicNumerals() {
```

Comment:
I personally do not like junit parametrized tests because every test method has to be in its own class and you have to a lot of boiler plate. For parametrized test I use https://github.com/TNG/junit-dataprovider

Theories can be nice for testing methods with multiple parameters, although in my opinion well chosen parameter combinations are often enough. I think theories are a bit lacking compared to "real" property based testing like quick-check (no test data generation, no failure shrinking). For java https://github.com/pholser/junit-quickcheck is the best option I'm aware of, but it has been requiring java 8 for years.